src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -132,115 +230,867 @@ static DamageTable D_8092C140 = {
+    if (ABS_ALT(temp_v0) <= 0x4000) {
+        Animation_MorphToPlayOnce(&this->skelAnime, &gIronKnuckleFrontHitAnim, -4.0f);
+    } else {
+        Animation_MorphToPlayOnce(&this->skelAnime, &gIronKnuckleBackHitAnim, -4.0f);
+    }
+    this->actionFunc = EnIk_ReactToAttack;
+}
+
+void EnIk_ReactToAttack(EnIk* this, GlobalContext* globalCtx) {
+    Math_StepToF(&this->actor.speedXZ, 0.0f, 1.0f);
+    if (this->subCamId != CAM_ID_MAIN) {
+        Play_CameraSetAtEye(globalCtx, this->subCamId, &this->actor.focus.pos,
+                            &Play_GetCamera(globalCtx, this->subCamId)->eye);
+    }
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (this->subCamId != CAM_ID_MAIN) {
+            ActorCutscene_Stop(this->actor.cutscene);
+            this->subCamId = CAM_ID_MAIN;
+            EnIk_SetupIdle(this);
+        } else {
+            EnIk_CheckActions(this, globalCtx);
+        }
+    } else {
+        this->actor.colorFilterTimer = 12;
+        this->invincibilityFrames = 12;
+    }
+}
+
+void EnIk_SetupDeath(EnIk* this) {
+    this->actor.speedXZ = 0.0f;
+    Animation_MorphToPlayOnce(&this->skelAnime, &gIronKnuckleDeathAnim, -4.0f);
+    this->counter = 24;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_DEAD);
+    this->actionFunc = EnIk_Death;
+}
+
+void EnIk_Death(EnIk* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f effectPos;
+
+    this->invincibilityFrames = 12;
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (this->counter == 24) {
+            SoundSource_PlaySfxEachFrameAtFixedWorldPos(globalCtx, &this->actor.world.pos, 35,
+                                                        NA_SE_EN_COMMON_EXTINCT_LEV - SFX_FLAG);
+        }
+
+        if (this->counter != 0) {
+            this->counter--;
+            for (i = 6 - (this->counter >> 2); i >= 0; i--) {
+                effectPos.x = randPlusMinusPoint5Scaled(80.0f) + this->actor.world.pos.x;
+                effectPos.z = randPlusMinusPoint5Scaled(80.0f) + this->actor.world.pos.z;
+                effectPos.y = randPlusMinusPoint5Scaled(50.0f) + (this->actor.world.pos.y + 20.0f);
+                func_800B3030(globalCtx, &effectPos, &sEffectVelAndAccel, &sEffectVelAndAccel, 100, 0, 2);
+            }
+            if (this->counter == 0) {
+                Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0xB0);
+                ActorCutscene_Stop(this->actor.cutscene);
+                Actor_MarkForDeath(&this->actor);
+            }
+        }
+        return;
+    }
+    if (Animation_OnFrame(&this->skelAnime, 23.0f)) {
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_WALK);
+    }
+}
+
+void EnIk_SetupFrozen(EnIk* this) {
+    this->invincibilityFrames = 0;
+    this->actionFunc = EnIk_Frozen;
+    this->actor.speedXZ = 0.0f;
+}
+
+void EnIk_Frozen(EnIk* this, GlobalContext* globalCtx) {
+    if (this->counter) {
+        this->counter--;
+    }
```

Comment:
wat. `if (this->counter)` probably shouldn't be simplified as its an int not a bool..

Is that too old fashioned?

Reply:
That should be `!= 0`, yes.